agent/xds/clusters.go

```diff
@@ -190,10 +196,13 @@ func makePassthroughClusters(cfgSnap *proxycfg.ConfigSnapshot) ([]proto.Message,
 				},
 				LbPolicy: envoy_cluster_v3.Cluster_CLUSTER_PROVIDED,
 
-				// TODO(tproxy) This should use the connection timeout configured on the upstream's config entry
```

Comment:
Should also remove the associated known limitation from the docs (third bullet):
https://www.consul.io/docs/connect/transparent-proxy#known-limitations

Though only the 5s portion, since the upstream's protocol is still not being considered.

Reply:
Actually the caveat should be updated to say that it can be specified via the service-resolver of the upstream but not upstream defaults:
https://www.consul.io/docs/connect/config-entries/service-defaults#upstreamconfig